src/main/java/com/fasterxml/jackson/datatype/jdk8/OptionalSerializer.java

```diff
@@ -315,6 +315,7 @@ public void acceptJsonFormatVisitor(JsonFormatVisitorWrapper visitor, JavaType t
             }
             _dynamicSerializers = _dynamicSerializers.newWith(type, ser);
         }
+        ser = (JsonSerializer<Object>)provider.handlePrimaryContextualization(ser, _property);
```

Comment:
When I was researching the issue of the Date format annotations not being honored, I used the java debugger to step through Jackson serialization without using Optional (and thus not using the jdk8 module) and then debugged/stepped through with Optional<Date> and compared the differences.  I tried to mimic the location/call pattern of the missing handlePrimaryContextualization as close as possible to how Jackson core does it.

Do we have some kind of performance suite to measure performance impact of this change?